---
title: "Example3_Treutlein"
author: "Nan Papili Gao"
date: "3/4/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## * CALISTA MAIN SCRIPT *

Example 3.Mouse embryonic fibroblast differentiation into neurons.
Analysis of RNA-seq data in Treutlein et al., [(Dissecting direct reprogramming from fibroblast to neuron using single-cell RNA-seq, Nature 534, 391–395 (2016)).](https://www.nature.com/articles/nature18323)

First download the example datasets from the following github [**link**](https://github.com/CABSEL/calista_example_data)

**Please unzip the file “3-TREUTLEIN_data_type_3_format_data_5_clusters_4.txt.zip” in EXAMPLES/TREUTLEIN/ before running CALISTA**

Please check [here](https://github.com/CABSEL/calista_R_old_v1.0.2/blob/master/CALISTA-R/1%20CALISTA_USER_MANUAL-R.pdf) (from page 14), for more info about how manually import from text file.


Use the following lines to run CALISTA on UNZIPPED Treutlein dataset (available in the subfolder EXAMPLES/TREUTLEIN):

```{r pressure, echo=TRUE,eval=FALSE}
rm(list=ls())
graphics.off()
options(warn=-1)
library(calista)
##########################################################################
# **Specify data types and settings for pre-processing**
INPUTS=list()
INPUTS$data_type=3 
# 1- for scRT-qPCR  (CT values)
# 2- for scRT-qPCR  (Expression values - Num. mRNA molecules)
# 3- for scRNA-seq  (Expression values - e.g log(TPM+1) or log(RPKM+1))
# 4- for scRNA-seq  (Expression values - e.g TPM+1 or RPKM)

INPUTS$format_data=5 
# 1- Rows= cells and Columns= genes with time/stage info in the last column  
# 2- Rows= genes and Columns= cells with time/stage info in the first row
# 3- Rows= cells and Columns= genes without time info
# 4- Rows= genes and Columns= cells without time info
# 5- Manual selection from data table

INPUTS$data_selection=integer() 
# When cells come from different capture times or stages, users can select 
# cells from specific time or stage for further analysis by CALISTA. 
# For instance, considering a dataset with cells taken from 4 time points, 
# snapshots can be selected, as follows:
# INPUTS$data_selection = integer() or c(1:4) for all time points/cells 
# INPUTS$data_selection = 1          for cells the first time point
# INPUTS$data_selection = c(1 3 4)    for cells 1st, 3rd and 4th time points

INPUTS$perczeros_genes=90 
# Users can exclude genes with a certain percentage of zeros. 
# INPUTS$perczeros_genes = 100 (recommended)

INPUTS$perczeros_cells=100 # Remove cells with 100% of zeros
# Users can exclude cells with more than a certain percentage of zeros.
# INPUTS$perczeros_cells = 100 (recommended)

INPUTS$cells_2_cut=0 
# Users can exclude cells from further analysis. 
# 1- Remove cells based on their indices in the expression matrix. 
#    Indices need to be uploaded as a separate csv file. 
# 0- No cells deletion

INPUTS$perc_top_genes=10 
# Retain only top X the most variable genes with X=min(200, INPUTS.perc_top_genes * num of cells/100, num of genes)
##########################################################################
# **Specify single-cell clustering settings**
INPUTS$optimize=1 
# 1- select the number of clusters by eigengap plot
# 0- define the number of clusters 

INPUTS$parallel=1
# 1- Use parallel processing (number of cores available - 1)  
# 0- Do not use processing

INPUTS$runs=50 
# Number of clustering runs (for greedy algorithm)
# INPUTS$runs = 50; 

INPUTS$max_iter=100 
# Maximum number of iterations in the greedy algorithm 
# INPUTS$max_iter = 100; 

INPUTS$Cluster='kmedoids' 
# 'hierarchical'- hierachical clustering of consensus matrix
# 'kmedoids'-  kmedoids for the clustering of consensus matrix

##########################################################################
# **Specify transition genes settings**
INPUTS$thr_transition_genes=75 
# Set threshold for transition genes determination to 50%

##########################################################################
# **** CALISTA ****
calista(INPUTS)

```
1. From the popup window select the dataset **3-TREUTLEIN_data_type_3_format_data_5_clusters_4.txt** in EXAMPLE/TREUTLEIN
```{r echo=FALSE,eval=TRUE}
writeLines('[1] "**** Please upload normalized data. File formats accepted: .txt , .xlxs , .csv ****"')

writeLines('Data loading...')
```
2. Manually import the data:
```{r echo=FALSE,eval=TRUE}
writeLines('Press 1 if columns=genes, 0 otherwise (e.g. 1):')
1
writeLines('Key the starting column for expression data (e.g. 6):')
6
writeLines('* Key starting and ending rows for the expression values (e.g. 1 405):*')
c(1, 405)
writeLines('* Key starting and ending columns for the expression values (e.g. 1 22524):*')
c(1, 22524)
writeLines('*Key the starting and ending columns for gene names(e.g. 6 22529)*')
c(6, 22529)
writeLines('*Add time info (1-Yes,0-No):*')
1
writeLines('*Key the column or row (e.g 5) vector in the raw data set with time/cell stage info:*')
5
```
2. In this case, the number of clusters is determined using the eigengap plot. According to the eigengap plot below, we set the number of clusters to **4** in the console:
```{r echo=FALSE,eval=TRUE}
writeLines('CALISTA_clustering is running...')

writeLines('Optimal number of cluster according to max. eigenvalue: 4 
if you want to use this value press 0, else provide desired number of cluster:')
4
```
3. If desired, users can remove cells from specific clusters from further analysis. In this example, we do not want to remove any clusters. Hence, we enter 0 (no cluster removal) and then 1 to proceed with lineage inference:
```{r echo=FALSE,eval=TRUE}
writeLines('Press 1 if you want to remove cell cluster(s). Press 0 otherwise: ')
0
writeLines('Press 1 if you want to perform additional analysis (e.g. lineage inference, cell ordering) , 0 otherwise: ')
1
```
4. Use CALISTA Shiny App to visualize the results (see video tutorial).
